From 4b23fc819d66e6163b62356d56a61db7276db81c Mon Sep 17 00:00:00 2001
From: Sridhar Nuvusetty <Sridhar.Nuvusetty@imgtec.com>
Date: Thu, 14 Apr 2016 17:37:47 +0530
Subject: UPSTREAM: mac80211: Requeue work after scan complete for all VIF
 types.

During a sw scan ieee80211_iface_work ignores work items for all vifs.
However after the scan complete work is requeued only for STA, ADHOC
and MESH iftypes.

This occasionally results in event processing getting delayed/not
processed for iftype AP when it coexists with a STA. This can result
in data halt and eventually disconnection on the AP interface.

Change-Id: I9eae2f033e8f07bc08b86721894ddad042e8490d
Signed-off-by: Sridhar Nuvusetty <Sridhar.Nuvusetty@imgtec.com>
Signed-off-by: Govindraj Raja <Govindraj.Raja@imgtec.com>
---
 net/mac80211/mesh.c | 1 -
 net/mac80211/mlme.c | 2 +-
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/net/mac80211/mesh.c b/net/mac80211/mesh.c
index 6f85b6a..d3b858f 100644
--- a/net/mac80211/mesh.c
+++ b/net/mac80211/mesh.c
@@ -1370,7 +1370,6 @@ out:
 	sdata_unlock(sdata);
 }
 
-
 void ieee80211_mesh_init_sdata(struct ieee80211_sub_if_data *sdata)
 {
 	struct ieee80211_if_mesh *ifmsh = &sdata->u.mesh;
diff --git a/net/mac80211/mlme.c b/net/mac80211/mlme.c
index 07051d5..077c8b7 100644
--- a/net/mac80211/mlme.c
+++ b/net/mac80211/mlme.c
@@ -1650,7 +1650,7 @@ void ieee80211_dynamic_ps_rx_recalc_ps_work(struct work_struct *work)
 			     dynamic_ps_rx_recalc_ps_work);
 
 	mutex_lock(&local->iflist_mtx);
-	ieee80211_recalc_ps(local, -1);
+	ieee80211_recalc_ps(local);
 	mutex_unlock(&local->iflist_mtx);
 }
 
-- 
1.9.1

